lib/services/osmdata.dart

```diff
@@ -183,21 +183,42 @@ class OsmData{
     }
     if(element['type'] == 'way'){
       double wayPenalty;
-      if(RegExp(r"footway|cyclepath|track|path|residential|unclassified|service").hasMatch(element['tags']['highway'])) {
+      if(RegExp(r"motorway|trunk|primary|motorway_link|trunk_link|primary_link").hasMatch(element['tags']['highway'])) {
+        wayPenalty = 20;
+      }
+      else if(RegExp(r"secondary|tertiary|secondary_link|tertiary_link").hasMatch(element['tags']['highway'])) {
+        wayPenalty = 8;
+      }
+      else if(RegExp(r"cyclepath|track|path|bridleway|sidewalk|residential|service").hasMatch(element['tags']['highway'])) {
+        wayPenalty = 2;
+      }
+      else if(RegExp(r"footway|pedestrian|unclassified").hasMatch(element['tags']['highway'])) {
         wayPenalty = 1;
       } else{
-        wayPenalty = 2;
+        wayPenalty = 5;
       }
       ways.add(Way(element['id'], element['nodes'].cast<int>(), this, wayPenalty));
     }
   }
 
-  //Todo: improve this distance function. Right now it assumes the earth is flat (which might be true).
   //http://edwilliams.org/avform.htm#Dist
   static double getDistance(Node nodeA, Node nodeB){
-    var a = (nodeA.latitude - nodeB.latitude).abs();
-    var b = (nodeA.longitude - nodeB.longitude).abs();
-    return sqrt(a*a + b*b);
+    //optimized haversine formular from https://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula
+    var p = 0.017453292519943295;    // PI / 180
```

Comment:
dart:math has pi constant or is it faster to use an precalculated shorter version? (math.pi)

Reply:
Yes, its supposed to be faster.